nuxeo-core/nuxeo-core-convert/src/main/java/org/nuxeo/ecm/core/convert/service/ConversionServiceImpl.java

```diff
@@ -226,52 +226,66 @@ public static String getCacheBasePath() {
+    }
+
+    protected BlobHolder convertThroughHTML(BlobHolder blobHolder, String destMimeType) {
+        Blob blob = blobHolder.getBlob();
         String mimetype = blob.getMimeType();
         String filename = blob.getFilename();
-        if (MimetypeRegistry.PDF_MIMETYPE.equals(mimetype)) {
-            return blob;
+        if (destMimeType.equals(mimetype)) {
+            return blobHolder;
         }
-        Blob result;
-        if (MediaType.TEXT_PLAIN.equals(mimetype)) {
-            result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
-        } else {
-            // Convert the blob to HTML
-            if (!MediaType.TEXT_HTML.equals(mimetype)) {
-                blob = convertBlobToMimeType(blob, MediaType.TEXT_HTML);
-                blob.setFilename(filename);
-            }
-            Path tempDirectory = Framework.createTempDirectory("blobs");
-            try {
-                // Replace the image URLs by absolute paths
-                DownloadService downloadService = Framework.getService(DownloadService.class);
-                blob = replaceURLsByAbsolutePaths(blob, tempDirectory, downloadService::resolveBlobFromDownloadUrl);
-                // Convert the blob to PDF
-                result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
-            } finally {
+
+        Path tempDirectory = null;
+        // Convert the blob to HTML
+        if (!MediaType.TEXT_HTML.equals(mimetype)) {
+            blobHolder = convertBlobToMimeType(blobHolder, MediaType.TEXT_HTML);
+        }
+        try {
+            tempDirectory = Framework.createTempDirectory("blobs");
+            // Replace the image URLs by absolute paths
+            DownloadService downloadService = Framework.getService(DownloadService.class);
+            blobHolder.setBlob(
+                    replaceURLsByAbsolutePaths(blob, tempDirectory, downloadService::resolveBlobFromDownloadUrl));
+            // Convert the blob to the destination mimetype
+            blobHolder = convertBlobToMimeType(blobHolder, destMimeType);
+            adjustBlobName(filename, blobHolder, destMimeType);
+        } catch (IOException e) {
+            throw new ConversionException(e);
+        } finally {
+            if (tempDirectory != null) {
                 org.apache.commons.io.FileUtils.deleteQuietly(tempDirectory.toFile());
             }
         }
-        if (result != null) {
-            adjustPDFBlobName(filename, result);
-        }
-        return result;
+        return blobHolder;
     }
 
-    protected Blob convertBlobToMimeType(Blob blob, String destinationMimeType) {
-        BlobHolder bh = new SimpleBlobHolder(blob);
-        bh = convertToMimeType(destinationMimeType, bh, null);
-        return bh == null ? null : bh.getBlob();
+    protected BlobHolder convertBlobToMimeType(BlobHolder bh, String destinationMimeType) {
+        return convertToMimeType(destinationMimeType, bh, null);
```

Comment:
Give `Collections.singletonMap()` for `parameters` parameter, as the underlining method could call `convert` which instantiate `parameters` as a `HashMap` if it's null.